src/test/regress/expected/multi_subquery_in_where_reference_clause.out

```diff
@@ -87,7 +87,7 @@ DETAIL:  Functions are not allowed in FROM clause when the query has subqueries
 SELECT
   user_id
 FROM
-  (SELECT  5 AS user_id) users_reference_table
+  (SELECT  5 AS user_id UNION ALL SELECT 6) users_reference_table
```

Comment:
Same here, if you can find a query where `HasEmptyJoinTree` returns true instead of `SELECT  5 AS user_id`, you wouldn't need to add a UNION.